go-controller/pkg/node/openflow_manager.go

```diff
@@ -275,12 +275,12 @@ func checkPorts(netConfigs []*bridgeUDNConfiguration, physIntf, ofPortPhys strin
 
 		}
 		if netConfig.ofPortPatch != curOfportPatch {
-			if netConfig.isDefaultNetwork() || curOfportPatch != "" {
+			if netConfig.isDefaultNetwork() {
```

Comment:
Just to highlight this for the future...the check ports is here in case ovn-controller has a bug and recreates the port:

https://github.com/openshift/ovn-kubernetes/commit/0b04aa4f4f7310f402cb65f3fe62317a5fe33ce5

"it so happens that ovn-controller, when it hits a snag, it can tear
down this patch cable under the foot of ovnkube-node and add it back.
this changes the `ofport` number and all that openflow rules are now
invalid"

This commit was from a long time ago, perhaps this is no longer a concern. I haven't seen it happen in years as a real failure. I think it is safe to ignore this for  now with UDNs, if we ever see a real bug in ovn from this then we can refer back to this comment and see if we want to add the functionality back.

Reply:
Thanks for the context!
I do think that even if there is a bug in ovn-controller in the future we should not crash ovn-k because some  UDN failed, we could certainly try to be more vocal through events as I suggested above but that can be done "in the future".